### ISSUE TITLE
question about the principle of the method

### ISSUE THREAD

**Issue by CoinCheung** [facebookresearch/FixRes#25]
Hi,

Thanks for releasing this great work !! after reading the paper and going through the code, I feel I have not fully understand the method ...

Would you please tell me what is the general pipeline of using the proposed method? Take resnet50 as instance, do I need to train resnet-50 with the standard method (train cropsize=224, test cropsize=224/256, achieve 76% acc), and then finetune it with larger cropsize of 384 for a few epoch and test with the enlarged cropsize of 384(resized the short side to 384 * 256 / 224 before crop) ? Or do I need to train and test with larger cropsize of 384 from scratch directly? Do we have other things to pay attention to except the training/test cropsizes ?

**Comment by TouvronHugo** [comment]
Hi @CoinCheung ,
The process is as follows:
1) Train ResNet-50 (resolution 224) with standard data-augmentation.
2) Fine-tune the last batch-norm + classifier with test time data-augmentation (see [transform_v2.py](https://github.com/facebookresearch/FixRes/blob/master/transforms_v2.py#L185)) and resolution 384.

I hope I have answered your question, don't hesitate if you have other questions.

Best,
Hugo

**Comment by CoinCheung** [comment]
Hi,

There are still some details that I am not sure about, First, in the training process, both training from scratch and finetune use standard augmentation defined in the torchvision library, and in the test process, they use self-defined `Resize` function, which resize the longer side of the image to assigned value when `largest=True`, and only resize the **hight** of the image to the assigned value when `largest=False`, why always resize the **hight** of the image to the assigned value? 
Second, in the `imnet_finetune/` folder, there is a `transform.py` which is actually used in the finetuning process. Do we need to replace it with the `transform_v2.py` ? Additionally, I found that when finetune the efficientnet, the last `nn.Liear` and `nn.Conv2d` is tuned, but the last `bn` layer is not set to `requires_grad=True`, meanwhile the state of the `bn` layer is set to `train` mode. What is the necessity of not tuning the `weight` of the last bn but still updating its `moving average states` ?

By the way, it seems that when finetune the resnet-50 model, all the parameters are tuned while finetuning the efficentnet needs only to finetune the last few layers. Did I misunderstand the code ?

**Comment by TouvronHugo** [comment]
Hi,
The resize is not important, just make sure to use the same one for testing and finetuning (see [issues4](https://github.com/facebookresearch/FixRes/issues/4)).
Yes you can do it, normally `transform_v2.py` gives slightly better results.
Excellent point, originally I didn't relearn the weights and biases of the last batch norm but only updated its statistics to adapt the network to the change in distribution due to the resolution change. I think you can  set `requires_grad=True` for `bn` it should not degrade performance.
For ResNet-50 we finetune the last bn and fully connected layer ([see](https://github.com/facebookresearch/FixRes/blob/master/imnet_finetune/train.py#L181)).
For finetuning it is sometimes interesting to finetune more layers to improve performance.

Best,

Hugo

**Comment by CoinCheung** [comment]
Hi, thanks for replying !!!  
So is the major difference between scratch training and finetuning is that we use smaller cropsize (both train/eval) to train from scratch and then use larger cropsize to finetune with backbone features freezed ? And as for the data augmentation of scratch training and finetuning, the standard implementation is good enough and we need only take care of the cropsizes, is that correct ?

**Comment by TouvronHugo** [comment]
Yes it is exactly that
